src/main/java/com/algorand/algosdk/v2/client/Utils.java

```diff
@@ -1,19 +1,179 @@
+     *                         include
+     * @return DryrunRequest to be submitted to TealDryrun endpoint 
+     * @throws Exception if transaction is rejected or the transaction is not
+     *                   confirmed before wait round
+     */
+    public static DryrunRequest createDryrun(AlgodClient client, List<SignedTransaction> txns, String protocol_version,
+            Long latest_timestamp, Long round)
+            throws Exception {
+
+        if (client == null || txns.size() == 0) {
+            throw new IllegalArgumentException("Bad arguments for createDryrun.");
+        }
+
+        // The details we need to add to DryrunRequest object
+        ArrayList<Application> app_infos = new ArrayList<>();
+        ArrayList<Account> acct_infos = new ArrayList<>();
+
+        // These are populated from the transactions passed
+        Set<Long> apps = new HashSet<>();
+        Set<Long> assets = new HashSet<>();
+        Set<String> accts = new HashSet<>();
+
+        for (SignedTransaction txn : txns) {
+            Transaction tx = txn.tx;
+
+            // We're only interested to pull state for app calls
+            if (tx.type != Type.ApplicationCall) {
+                continue;
+            }
+
+            // If this is a create transaction
+            if (tx.applicationId == 0 || tx.applicationId == null) {
+                // Prepare and set param fields for Application being created
+                // from the transaction passed
+                ApplicationParams params = new ApplicationParams();
+                params.creator = tx.sender;
+                params.approvalProgram = tx.approvalProgram.getBytes();
+                params.clearStateProgram = tx.clearStateProgram.getBytes();
+
+                ApplicationStateSchema localState = new ApplicationStateSchema();
+                localState.numByteSlice = tx.localStateSchema.numByteSlice.longValue();
+                localState.numUint = tx.localStateSchema.numUint.longValue();
+                params.localStateSchema = localState;
+
+                ApplicationStateSchema globalState = new ApplicationStateSchema();
+                globalState.numByteSlice = tx.globalStateSchema.numByteSlice.longValue();
+                globalState.numUint = tx.globalStateSchema.numUint.longValue();
+                params.globalStateSchema = globalState;
+
+                Application app = new Application();
+                app.id = Utils.defaultAppId;
+                app.params = params;
+
+                app_infos.add(app);
+            } else {
+                apps.add(tx.applicationId);
+                accts.add(Address.forApplication(tx.applicationId).toString());
+            }
+
+            if (tx.foreignApps.size() > 0) {
+                apps.addAll(tx.foreignApps);
+            }
+
+            if (tx.foreignAssets.size() > 0) {
+                assets.addAll(tx.foreignAssets);
+            }
+
+            if (tx.accounts.size() > 0) {
```

Comment:
I saw in implementation of createDryrun for other SDK that you add `sender` to `accts`, wondering do we also want to do this in Java, or `sender` is included in `tx.accounts` so we can omit this.